File: src/main/scala/examples/lambdas/Example2.scala
package examples.lambdas

class Example2 {
  def useIt(): Unit = {
    println(condition(true, helloWorld))
  }

  def condition(condition: Boolean, f: () => String): String = {
    if (condition)
      f.apply()
    else
      "false"
  }

  def helloWorld(): String =
    "Hello Java 8"
}

File: src/main/java/controls/lambdas/Example2.java
package controls.lambdas;

import java.util.function.Supplier;

public class Example2 {
    @SuppressWarnings("Convert2MethodRef")
    public void useIt() {
        System.out.println(condition(true, () -> this.helloWorld()));
    }

    public void useItWithMethodReference() {
        System.out.println(condition(true, this::helloWorld));
    }

    public String condition(boolean condition, Supplier<String> f) {
        if (condition) {
            return f.get();
        } else {
            return "false";
        }
    }

    public String helloWorld() {
        return "Hello Java 8";
    }
}
